modin/experimental/core/execution/native/implementations/omnisci_on_native/exchange/dataframe_protocol/utils.py

```diff
@@ -0,0 +1,53 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Utility functions of DataFrame exchange protocol implementation for OmnisciOnNative execution."""
+
+import pyarrow as pa
+import numpy as np
+
+from modin.core.dataframe.base.exchange.dataframe_protocol.utils import (
+    ArrowCTypes,
+    pandas_dtype_to_arrow_c,
+)
+
+
+def arrow_dtype_to_arrow_c(dtype: pa.DataType) -> str:
+    """
+    Represent PyArrow `dtype` as a format string in Apache Arrow C notation.
+
+    Parameters
+    ----------
+    dtype : pa.DataType
+        Datatype of PyArrow table to represent.
+
+    Returns
+    -------
+    str
+        Format string in Apache Arrow C notation of the given `dtype`.
+    """
+    if pa.types.is_timestamp(dtype):
+        return ArrowCTypes.TIMESTAMP.format(
+            resolution=dtype.unit[:1], tz=dtype.tz or ""
+        )
+    elif pa.types.is_date(dtype):
+        return getattr(ArrowCTypes, f"DATE{dtype.bit_width}", "DATE64")
```

Comment:
This line returns `DATE{bit_width}`, so this would return a proper format string for according type (date32 or date64). If the `bit_width` is out of {32, 64} set, return DATE type with maximum capacity (DATE64).

Reply:
Ok, I see.